src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -630,6 +633,97 @@ private static JsValue ConvertJsTimeToTimeSpanString(JsValue self, JsValue[] arg
                 return new JsValue(asTimeSpan.ToString());
             }
 
+            private static JsValue ToStringWithFormat(JsValue self, JsValue[] args)
+            {
+                if (args.Length < 2 || args.Length > 3)
+                {
+                    throw new InvalidOperationException($"No overload for method 'toStringWithFormat' takes {args.Length} arguments. " +
+                                                        "Supported overloads are : toStringWithFormat(object, format), toStringWithFormat(object, culture), toStringWithFormat(object, format, culture).");
+                }
+
+                CultureInfo cultureInfo = null;
+                string format = null;
+
+                for (var i = 1; i < args.Length; i++)
+                {
+                    if (args[i].IsString() == false)
+                    {
+                        throw new InvalidOperationException("toStringWithFormat : 'format' and 'culture' must be string arguments");
+                    }
+
+                    GetFormatOrCulture(args[i].AsString(), ref format, ref cultureInfo);
+                }
+
+                if (args[0].IsDate())
+                {
+                    var date = args[0].AsDate().ToDateTime();
+                    if (format != null && cultureInfo != null)
+                    {
+                        return date.ToString(format, cultureInfo);
+                    }
+                    return format != null ? 
+                        date.ToString(format) : 
+                        date.ToString(cultureInfo);
+
+                }
+
+                if (args[0].IsNumber())
+                {
+                    var num = args[0].AsNumber();
+                    if (format != null && cultureInfo != null)
+                    {
+                        return num.ToString(format, cultureInfo);
+                    }
+                    return format != null ?
+                        num.ToString(format) :
+                        num.ToString(cultureInfo);
+                }
+
+                if (args[0].IsBoolean() == false)
+                {
+                    throw new InvalidOperationException($"toStringWithFormat() is not supported for objects of type {args[0].Type} ");
+                }
+
+                var boolean = args[0].AsBoolean();
+                return boolean.ToString(cultureInfo);
+            }
+
+            private static void GetFormatOrCulture(string arg, ref string format, ref CultureInfo cultureInfo)
+            {
+                var split = arg.Split(".");
+                if (split.Length < 2 ||
+                    split[0] != nameof(CultureInfo))
+                {
+                    format = arg;
+                }
+                else
+                {
+                    switch (split[1])
+                    {
+                        case "CurrentCulture":
```

Comment:
I think that we want to support actual culture over the wire with InvariantCulture being `null` or empty string `''`

e.g.
`x.Start.ToString(CultureInfo.InvariantCulture)` => `toStringWithFormat(new Date(Date.parse(x.Start)))`
`x.Start.ToString(CultureInfo.InvariantCulture)` => `toStringWithFormat(new Date(Date.parse(x.Start)), '')`
`x.Start.ToString(CultureInfo.InvariantCulture)` => `toStringWithFormat(new Date(Date.parse(x.Start)), null)`

and when we have a different culture (let say `en-US`)
`x.Start.ToString(CultureInfo.CurrentCulture)` => `toStringWithFormat(new Date(Date.parse(x.Start)), 'en-US')`

Reply:
done